src/com/xilinx/rapidwright/util/CodeGenerator.java

```diff
@@ -174,4 +187,249 @@ public static void addPIPs(Net net, String[] pips) {
             net.addPIP(device.getPIP(pip));
         }
     }
+
+    /**
+     * Shorthand method to route site wires with a particular net in a site
+     * instance.
+     * 
+     * @param si        The site instance to target
+     * @param n         The net to be site routed
+     * @param siteWires The list of site wire names to route
+     */
+    public static void routeSiteNet(SiteInst si, Net n, String... siteWires) {
+        for (String siteWire : siteWires) {
+            BELPin bp = si.getSiteWirePins(siteWire)[0];
+            si.routeIntraSiteNet(n, bp, bp);
+        }
+    }
+    
+    /**
+     * Shorthand method for turning on site PIPs in a site instance.
+     * 
+     * @param si       The target site instance.
+     * @param sitePIPs The list of site PIPs (of the format <BEL name>:<input pin
+     *                 name>) to turn on.
+     */
+    public static void addSitePIPs(SiteInst si, String... sitePIPs) {
+        for (String sitePIP : sitePIPs) {
+            int colonIdx = sitePIP.indexOf(':');
+            si.addSitePIP(sitePIP.substring(0, colonIdx), sitePIP.substring(colonIdx + 1));
+        }
+    }
+
+    /**
+     * Shorthand method for creating a placed cell inside a site instance. This is
+     * mostly useful for constructing test case scenarios.
+     * 
+     * @param si          The target site instance
+     * @param name        Name of the cell to create
+     * @param isRoutethru Flag indicating if the cell should be a routethru
+     * @param type        The type of the cell (Unisim or special field)
+     * @param bel         The name of the BEL where the cell should be placed
+     * @param pinMaps     A variable number of pin mappings for the cell where each
+     *                    string is of the format <physical pin name>:<logical pin
+     *                    name>
+     * @return The created cell.
+     */
+    public static Cell genCell(SiteInst si, String name, boolean isRoutethru, String type, String bel,
+            String... pinMaps) {
+        Cell c = null;
+        if (isRoutethru || type.startsWith("<")) {
```

Comment:
Did you find `type.startsWith("<")` to be necessary, which may imply that we aren't setting `Cell.isRoutethru()` correctly?

Reply:
We have several "special" cells that aren't routethrus but don't have logical equivalence either.  They all have a special type that begins with "<", those are ones we want to detect here.